test/test_xbuffer.cpp

```diff
@@ -0,0 +1,82 @@
+/***************************************************************************
+* Copyright (c) QuantStack                                                 *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+***************************************************************************/
+
+#include "doctest/doctest.h"
+
+#include <numeric>
+
+#include "xparrow/xbuffer.hpp"
+
+namespace xp
+{
+    using buffer_test_type = xbuffer<uint8_t>;
+
+    namespace
+    {
+        uint8_t* make_test_buffer(std::size_t size, uint8_t start_value = 0)
+        {
+            uint8_t* res = new uint8_t[size];
+            std::iota(res, res + size, start_value);
+            return res;
+        }
+    }
+
+    TEST_SUITE("xbuffer")
+    {
+        TEST_CASE("constructors")
+        {
+            buffer_test_type b0;
+            CHECK_EQ(b0.data(), nullptr);
+            CHECK_EQ(b0.size(), 0u);
+
+            std::size_t size = 4u;
+            buffer_test_type b1(size);
+            CHECK_NE(b1.data(), nullptr);
+            CHECK_EQ(b1.size(), size);
+
+            uint8_t* mem = make_test_buffer(size);
+            buffer_test_type b2(mem, size);
+            CHECK_EQ(b2.data(), mem);
+            CHECK_EQ(b2.size(), size);
+            CHECK_EQ(b2.data()[2], uint8_t(2));
+        }
+
+        TEST_CASE("copy semantic")
+        {
+            std::size_t size = 4;
+            buffer_test_type b1(make_test_buffer(size), size);
+            buffer_test_type b2(b1);
+            CHECK_EQ(b1, b2);
+
+            std::size_t size2 = 8;
+            buffer_test_type b3(make_test_buffer(size2, 4), size2);
+            b2 = b3;
+            CHECK_EQ(b2, b3);
+            CHECK_NE(b1, b2);
+        }
+
+        TEST_CASE("move semantic")
+        {
+            std::size_t size = 4;
+            buffer_test_type b1(make_test_buffer(size), size);
+            buffer_test_type control(b1);
+            buffer_test_type b2(std::move(b1));
+            CHECK_EQ(b2, control);
+            CHECK_EQ(b1.size(), 0);
+            CHECK(b1.empty());
+            CHECK_EQ(b1.data(), nullptr);
+
+            std::size_t size2 = 8;
+            buffer_test_type b4(make_test_buffer(size2, 4), size2);
+            buffer_test_type control2(b4);
+            b2 = std::move(b4);
+            CHECK_EQ(b2, control2);
+            CHECK_EQ(b4, control);
+        }
+    }
```

Comment:
Please add tests for:
- equality comparison
- swaping
- a super basic test where the object is created and destroyed immediately
- another basic test wher the object is created, filled with values, then destroyed (these 2 are not equivalent to the constructor tests above)
- resizing
- basic checks (maybe in existing tests) that data written in a buffer is the data obtained by reading (in particular after moves and copies would be helpful in the future)